storage/namespace.go

```diff
@@ -467,29 +464,25 @@ func (n *dbNamespace) FlushState(blockStart time.Time) fileOpState {
 			continue
 		}
 		state := shard.FlushState(blockStart)
-		status := state.Status
-		if status == fileOpNotStarted || status == fileOpInProgress {
-			// Any not started or in progress overrides other states
-			return state
+		if state.Status == fileOpInProgress {
+			return false
 		}
-		if status == fileOpFailed {
+		if state.Status == fileOpNotStarted {
+			return true
+		}
+		if state.Status == fileOpFailed {
 			anyFailed = true
 			if state.NumFailures < minNumFailures {
```

Comment:
hm why not compare `state.NumFailures` with `n.opts.MaxFlushRetries` and if it's smaller, return true immediatley? Don't think we actually need the `anyFailed` flag